soapui/src/main/java/com/eviware/soapui/plugins/ProductBodyguard.java

```diff
@@ -0,0 +1,153 @@
+package com.eviware.soapui.plugins;
+
+import com.eviware.soapui.SoapUI;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.Provider;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Enumeration;
+import java.util.Vector;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+public final class ProductBodyguard extends Provider {
+    private X509Certificate providerCert = null;
+
+    public ProductBodyguard() {
+        super("SoapUIOSPluginSignChecker", 1.0, "The new plugin framework restriction");
```

Comment:
"The new plugin framework restriction" should be replaced by "Plugin signature validity checker", I think. It won't be new for long, and the restriction has always been there, really, only not technically.